src/pages/myRegisterdShow/components/registeredcard/RegisteredCard.tsx

```diff
@@ -24,40 +25,69 @@ const RegisteredCard = ({
   };
 
   const getShowTypeText = (key: SHOW_TYPE_KEY): ShowTypes => SHOW_TYPE[key];
+  const calculateDueDate = (dateString: string): number => {
+    //문자열이 ~ 를 포함하는지 확인 (즉, 단일 문자인지 아닌지 확인)
+    const endDateString = dateString.includes("~")
+      ? dateString.split("~")[1].trim()
+      : dateString.trim();
+
+    //해당 날짜를 표준 형식(YYYY-MM-DD) 형식으로 변환 (물론 YYYY.MM.DD도 인식되긴 함 -표준은 아님)
+    const [year, month, day] = endDateString.split(".").map(Number); //숫자로 저장
+    const endDate = new Date(year, month - 1, day); //month는 0부터 시작하므로 -1 해줌
+
+    //현재 시간을 얻은 뒤, 밀리초 계산은 배제하기 위해 연,월,일로 현재 날짜의 Date 객체 재생성
+    const nowDate = new Date();
+    const nowYear = nowDate.getFullYear();
+    const nowMonth = nowDate.getMonth();
+    const nowDay = nowDate.getDate(); //getDay는 요일을 반환하는거니까, 헷갈리지 말자 !!
+    const startDate = new Date(nowYear, nowMonth, nowDay);
+
+    //두 날짜간의 차이를 계산
+    const timeDiff = endDate.getTime() - startDate.getTime();
+
+    //계산값을 일 단위로 변환(floor를 이용해 내림)
+    const dayDiff = Math.floor(timeDiff / (1000 * 3600 * 24));
```

Comment:
매직넘버 상단에 따로 빼는게 어떨까요

```suggestion
    const A_DAY = 1000 * 3600 * 24
    // 네이밍 어렵다..
    // const ONE_DAY = 1000 * 3600 * 24 ? 가 낫나..
    const dayDiff = Math.floor(timeDiff / A_DAY);
```